pkg/cluster/gossip_test.go

```diff
@@ -117,27 +122,27 @@ func TestConfig_ApplyDefaults(t *testing.T) {
 		gc := defaultConfig
 		gc.DiscoverPeers = `provider=static addrs=fizzbuzz:5555`
 
-		err := gc.ApplyDefaults(examplePort)
+		err := gc.ApplyDefaults()
 		require.NoError(t, err)
-		require.Equal(t, []string{"fizzbuzz:5555"}, []string(gc.JoinPeers))
+		require.Equal(t, []string{"fizzbuzz:5555"}, getPeers(t, gc))
 	})
 
 	t.Run("peers can use default port", func(t *testing.T) {
 		gc := defaultConfig
-		gc.JoinPeers = []string{"foobar"}
+		gc.JoinPeers = []string{"192.168.1.14"}
```

Comment:
With the new behavior, `foobar` was being handled as a DNS record that cannot be looked up, so I changed it to an IP address.